src/Internal/Service/Blockfrost.purs

```diff
@@ -673,9 +677,38 @@ doesTxExist txHash = do
     Left e -> Left e
 
 --------------------------------------------------------------------------------
--- Get transaction metadata
---------------------------------------------------------------------------------
+-- Get transaction auxiliary data
+getTxAuxiliaryData
+  :: TransactionHash
+  -> BlockfrostServiceM (Either GetTxMetadataError AuxiliaryData)
+getTxAuxiliaryData txHash = runExceptT do
+  metadata <- ExceptT $ getTxMetadata txHash
+  (scriptRefs :: Array ScriptRef) <- ExceptT $ getTxScripts txHash
+  pure $ wrap
+    { metadata: Just metadata
+    , nativeScripts: arrayToMaybe $ getNativeScripts scriptRefs
+    -- , nativeScripts: Nothing
+    , plutusScripts: arrayToMaybe $ getPlutusScripts scriptRefs
+    }
+
+  where
+  arrayToMaybe :: forall a. Array a -> Maybe (Array a)
+  arrayToMaybe [] = Nothing
+  arrayToMaybe xs = Just xs
+
+  getNativeScripts :: Array ScriptRef -> Array NativeScript
+  getNativeScripts = catMaybes <<< map isNativeScript
+    where
+    isNativeScript (NativeScriptRef script) = Just script
+    isNativeScript (PlutusScriptRef _) = Nothing
 
+  getPlutusScripts :: Array ScriptRef -> Array PlutusScript
+  getPlutusScripts = catMaybes <<< map isPlutusScript
+    where
+    isPlutusScript (PlutusScriptRef script) = Just script
+    isPlutusScript (NativeScriptRef _) = Nothing
+
+-- TODO: refactor to aux and remove metadat from tests. add aux tests.
```

Comment:
Please implement `getTxMetadata` via `getTxAuxiliaryData` and add a deprecation warning for the former.

Reply:
Addressed in f72387c